icicle/include/poseidon2/kernels.cuh

```diff
@@ -2,10 +2,27 @@
 #ifndef POSEIDON2_KERNELS_H
 #define POSEIDON2_KERNELS_H
 
+#include "utils/utils.h"
+#include "hash/hash.cuh"
+#include "matrix/matrix.cuh"
 #include "poseidon2/constants.cuh"
 #include "gpu-utils/modifiers.cuh"
 
+using matrix::Matrix;
+
 namespace poseidon2 {
+  static DEVICE_INLINE unsigned int d_next_pow_of_two(unsigned int v)
+  {
+    v--;
```

Comment:
If `v` is a power of two, should this still return the next power of two or `v`?

Reply:
If `v` is a power of two, this should return `v`